### ISSUE TITLE
mpris: Attribute Error (CompositeObject) with Clementine

### ISSUE THREAD

**Issue by lasers** [ultrabug/py3status#911]
I can't bisect this one. I would guess something to do with the formatter changes.
```
lasers~/src/py3status/py3status/modules git:(master) python3 mpris.py.
Traceback (most recent call last):
  File "mpris.py", line 578, in <module>
    module_test(Py3status)
  File "/usr/lib/python3.6/site-packages/py3status-3.6rc1-py3.6.egg/py3status/module_test.py", line 47, in module_test
    module.post_config_hook()
  File "mpris.py", line 176, in post_config_hook
    self._start_listener()
  File "mpris.py", line 454, in _start_listener
    self._get_players()
  File "mpris.py", line 440, in _get_players
    self._add_player(player)
  File "mpris.py", line 411, in _add_player
    subscription = player.PropertiesChanged.connect(self._player_monitor(player_id))
AttributeError: '<CompositeObject>' object has no attribute 'PropertiesChanged'
```

**Comment by tobes** [comment]
> I can't bisect this one.

how to reproduce is more important than anything.



> /usr/lib/python3.6/site-packages/py3status-3.6rc1-py3.6.egg/py3status/module_test.py

This is not where I would expect you code to be running.  It looks like you did `python setup.py install` not `python setup.py develop`

**Comment by lasers** [comment]
I forget `develop`. I did `python setup.py install` and `python setup.py build`.

Using `install+develop` this time, I went back to a commit after mpris gets fixed two weeks ago (bf67bbec) and still get `AttributeError: <CompositeObject>` so I don't know where I went wrong again.

Now it shows...
>File "/home/chris/src/py3status/py3status/module_test.py", line 47, in module_test

```
lasers~/src/py3status/py3status/modules git:(bf67bbec) python2 mpris.py 
Traceback (most recent call last):
  File "mpris.py", line 567, in <module>
    module_test(Py3status)
  File "/home/chris/src/py3status/py3status/module_test.py", line 47, in module_test
    module.post_config_hook()
  File "mpris.py", line 165, in post_config_hook
    self._start_listener()
  File "mpris.py", line 443, in _start_listener
    self._get_players()
  File "mpris.py", line 429, in _get_players
    self._add_player(player)
  File "mpris.py", line 400, in _add_player
    subscription = player.PropertiesChanged.connect(self._player_monitor(player_id))
AttributeError: '<CompositeObject>' object has no attribute 'PropertiesChanged'
```

**Comment by tobes** [comment]
> Using install+develop this time

you know only one wins, they undo each other.

> how to reproduce 

any special format or does this happen with the default?  Does a player need to be running?  Any player or specific one?  Do you need to do something to trigger the issue?

**Comment by lasers** [comment]
I did some detective works.

>any special format or does this happen with the default?

No special format. I was giving `mpris` a go with empty/default `mpris {}` thing. I wasn't running a player. It came out broken too just like `clock` and I see `CompositeObject` bit so I guess it was related to recent formatter PR too.

>Does a player need to be running? Any player or specific one? Do you need to do something to trigger the issue?

Okay, I was unaware it had more to do with players than `Composite/formatter` stuffs since it came out broken too just like `clock`.

I found the issue. It turns out that I was probably unaware about running a player too as `clementine` was in the tray. I looked it up and it is a MPRIS1+2 compatible player. The bug is that it was trying to work with `clementine`.

**Comment by tobes** [comment]
ok so the issue is with `pydbus` and it's `CompositeObject` https://github.com/LEW21/pydbus/blob/f2f11a1ec41360de00b1d66a3a4a24884b6f2308/pydbus/proxy.py#L95

So the issue is we only support `mpris2`

**Comment by lasers** [comment]
Slight out of place, but it'll do here.

`mpris` trips with `on_click` when using `gpmdp` (Google Play Music Desktop Player). This is relevant because the issue is with `pydbus` too. 
```
 2017-11-16 INFO trying to dispatch event to module "mpris"
2017-11-16 WARNING on_click event in `mpris` failed (Error) mpris.py line 592.
2017-11-16 INFO Traceback
Error: g-io-error-quark: Method 'Pause' returned type '(u)', but expected '()' (13)
  File "/home/chris/src/py3status/py3status/module.py", line 687, in click_event
    click_method(event)
  File "/home/chris/src/py3status/py3status/modules/mpris.py", line 592, in on_click
    getattr(self._player, self._control_states[index]['action'])()
  File "/usr/lib/python3.6/site-packages/pydbus/proxy_method.py", line 75, in __call__
    0, timeout_to_glib(timeout), None).unpack()

2017-11-16 ERROR py3status: on_click event in `mpris` failed (Error) mpris.py line 592. Please try to fix this and reload i3wm
(Mod+Shift+R)
```

**Comment by AnwariasEu** [comment]
@tobes 
> ok so the issue is with pydbus and it's CompositeObject https://github.com/LEW21/pydbus/blob/f2f11a1ec41360de00b1d66a3a4a24884b6f2308/pydbus/proxy.py#L95
>
> So the issue is we only support mpris2

What exactly is the problem here? Clementine only supports mpris1 while the mpris module uses mpris2 for connect to the media players?

Would there be a way to backport functionality to mpris1?

**Comment by lasers** [comment]
Closed via #1648.
